File: test_spacing/nodejs_cli/cli_hooks.mjs
/**
 * Hook implementations for Spacing Test CLI
 * 
 * This file contains the business logic for your CLI commands.
 * Implement the hook functions below to handle your CLI commands.
 * 
 * IMPORTANT: Hook names must use snake_case with 'on_' prefix
 * Example:
 * - Command 'hello' -> Hook function 'on_hello'
 * - Command 'hello-world' -> Hook function 'on_hello_world'
 */

// Import any modules you need here
import fs from 'fs';
import path from 'path';
/**
 * Say hello with spacing * @param {string} name - Name to greet * @param {Object} options - Command options * @param {flag} options.verbose - Verbose output * @returns {Promise<number>} - Exit code (0 for success)
 */
async function on_hello(name, options) {    // Generic command implementation
    console.log('Executing hello command...');    console.log('Arguments received:', { name });    console.log('Options received:', options);    console.log('✅ Command completed successfully!');    
    return 0;
}
// Export all hook functions
export {    on_hello,};

File: test_spacing/src/cli_hooks.rs
// Hook implementations for Spacing Test CLI
//
// This file contains the business logic for your CLI commands.
// Implement the hook functions below to handle your CLI commands.
//
// IMPORTANT: Hook names must use snake_case with 'on_' prefix
// Example:
// - Command 'hello' -> Hook function 'on_hello'
// - Command 'hello-world' -> Hook function 'on_hello_world'

use clap::ArgMatches;
use anyhow::Result;

/// Say hello with spacing
pub fn on_hello(matches: &ArgMatches) -> Result<()> {
    // Add your business logic here
    println!("Executing hello command");
    // Access options
    if let Some(verbose) = matches.get_one::<String>("verbose") {
        println!("verbose: {}", verbose);
    }
    
    Ok(())
}